test/project.controller.test.js

```diff
@@ -0,0 +1,455 @@
+    });
+    it('should return 400 for invalid team data on POST', done => {
+        const invalidTeamData = {}; // Invalid or incomplete data
+        chai.request(server)
+            .post('/team')
+            .send(invalidTeamData)
+            .end((err, res) => {
+                expect(res).to.have.status(400);
+                done();
+            });
+    });
+
+    // PUT request with invalid data
+    it('should return 400 for invalid team data on PUT', done => {
+        const teamId = 1; // Existing team ID for testing
+        const invalidTeamData = {}; // Invalid or incomplete data
+        chai.request(server)
+            .put(`/team/${teamId}`)
+            .send(invalidTeamData)
+            .end((err, res) => {
+                expect(res).to.have.status(400);
+                done();
+            });
+    });
+
+    // DELETE request for a non-existent team
+    it('should return 404 for a non-existent team on DELETE', done => {
+        const teamId = 999; // Non-existent team ID
+        chai.request(server)
+            .delete(`/team/${teamId}`)
+            .end((err, res) => {
+                expect(res).to.have.status(404);
+                done();
+            });
+    });
+});
+
+// Error scenario tests for User Controller
+describe('User Controller Error Scenarios', () => {
+    it('should return 404 for a non-existent user', done => {
+        const userId = 999; // Non-existent user ID
+        chai.request(server)
+            .get(`/user/${userId}`)
+            .end((err, res) => {
+                expect(res).to.have.status(404);
+                done();
+            });
+    });
+
+    it('should return 400 for invalid user data on POST', done => {
+        const invalidUserData = {}; // Invalid or incomplete data
+        chai.request(server)
+            .post('/user')
+            .send(invalidUserData)
+            .end((err, res) => {
+                expect(res).to.have.status(400);
+                done();
+            });
+    });
+
+    // PUT request with invalid data
+    it('should return 400 for invalid user data on PUT', done => {
+        const userId = 1; // Existing user ID for testing
+        const invalidUserData = {}; // Invalid or incomplete data
+        chai.request(server)
+            .put(`/user/${userId}`)
+            .send(invalidUserData)
+            .end((err, res) => {
+                expect(res).to.have.status(400);
+                done();
+            });
+    });
+
+    // DELETE request for a non-existent user
+    it('should return 404 for a non-existent user on DELETE', done => {
+        const userId = 999; // Non-existent user ID
+        chai.request(server)
+            .delete(`/user/${userId}`)
+            .end((err, res) => {
+                expect(res).to.have.status(404);
+                done();
+            });
+    });
+});
+
+// Close the server after tests run
+after(done => {
+    server.close(() => {
+        done();
+    });
+});
```

Comment:
I would split this file to separate files based on specific controller that the tests target, so ... project.controller.test.js, team.controller.test.js etc.
What do you think?

Reply:
Sure! I'll tackle this tomorrow. One task I've identified (and am contemplating whether it requires a separate ticket or just a backlog request) is the lack of error handling for invalid data in our backend. This issue has led to some of our error scenario tests failing:

```
PS C:\developer\Task-Manager> npm test
...
23 passing (231ms)
8 failing
...
```
These failures mostly involve expected status codes of 400 not being returned for invalid data in POST and PUT requests across various controllers. (it could also be that the tests are not correctly written, I ll take a look :D) I'll make sure to either log a ticket or add this to our backlog by tomorrow. Also not sure whether you or @hajekad want to tackle this.

Reply:
Let me know

Reply:
I can do this, we need validation od input dto on all endpoints, and throw 400 on validation failure
I will try to implement and push it today @MartinPl0

Reply:
Thanks!!!!

Reply:
input dto validation implemented https://github.com/Project-team-DL-7/Task-Manager/pull/13